### PR TITLE
Port otel/1090

### PR DESCRIPTION
**Pull request by devinsba**
Ports https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1090 from @trask  which since I also believe this is not needed.

TODO add some tests with java 8 and lambdas to ensure this isn't breaking things subtly and add some tests around executors that are on our classpath that have their own queue implementations

### CODE REVIEW

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/ThreadPoolExecutorInstrumentation.java [hunk 1]
```diff
@@ -1,77 +0,0 @@
-package datadog.trace.instrumentation.java.concurrent;
```
Comment (devinsba) [inline dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/ThreadPoolExecutorInstrumentation.java:1]: @mcculls looks like you an I might have some conflict here

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/ExecutorInstrumentationUtils.java [hunk 2]
```diff
@@ -12,9 +11,6 @@
 @Slf4j
 public class ExecutorInstrumentationUtils {
 
-  private static final WeakMap<Executor, Boolean> EXECUTORS_DISABLED_FOR_WRAPPED_TASKS =
```
Comment (richardstartin) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/ExecutorInstrumentationUtils.java:15]: Aww I hate seeing these disappear :(

**Review by richardstartin — APPROVED** [review]

**Review by tylerbenson — APPROVED** [review]
I forget... without this, how is `ScheduledExecutorService` handled?

**Comment by devinsba** [comment]
The two additional methods were already covered: https://github.com/DataDog/dd-trace-java/blob/3118568d508f11fcd8d799db3a1115b81f4a316a/dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaExecutorInstrumentation.java#L69...L74

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/ThreadPoolExecutorInstrumentation.java (hunk 1 above)
Comment (devinsba) [reply dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/ThreadPoolExecutorInstrumentation.java:1]: @mcculls deleting this file seems to be the same thing you accomplished in your PR right? Removing the weakmap access?

**Pull request merged**